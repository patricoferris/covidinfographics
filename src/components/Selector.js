import React from 'react';
import { navigate } from "gatsby"
import { LocaleContext } from "./Layout"
import { makeStyles } from '@material-ui/core/styles';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import FormHelperText from '@material-ui/core/FormHelperText';
import Select from '@material-ui/core/Select';

const useStyles = makeStyles((theme) => ({
  formControl: {
    margin: theme.spacing(1),
    minWidth: 120,
  },
  selectEmpty: {
    marginTop: theme.spacing(2),
  },
}));

const Selector = ({ languages }) => {
  const classes = useStyles();
  const { locale } = React.useContext(LocaleContext)
  const [language, setLanguage] = React.useState(languages.find(lang => lang.path === locale));

  const handleChange = (event) => {
    const path = event.target.value.path === "en" ? "/" : `/${event.target.value.path}`
    setLanguage(event.target.value);
    navigate(path);
  };

  return (
    <FormControl className={classes.formControl}>
        <Select
          value={language}
          onChange={handleChange}
          displayEmpty
          className={classes.selectEmpty}
          inputProps={{ 'aria-label': 'Without label' }}
        >
          {languages.map(lang => <MenuItem value={lang}>{lang.name}</MenuItem>)}
        </Select>
        <FormHelperText>English</FormHelperText>
      </FormControl>
  )
}

export default Selector

