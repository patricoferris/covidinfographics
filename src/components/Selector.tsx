import React from 'react'
import { navigate } from 'gatsby'
import { LocaleContext } from './Layout'
import { makeStyles } from '@material-ui/core/styles'
import MenuItem from '@material-ui/core/MenuItem'
import FormControl from '@material-ui/core/FormControl'
import FormHelperText from '@material-ui/core/FormHelperText'
import Select from '@material-ui/core/Select'

interface LocaleProperpties {
  default: boolean
  path: string
  name: string
  english_name: string
  locale: string
  dateFormat: string
  siteLanguage: string
  ogLanguage: string
  defaultTitle: string
  defaultDescription: string
}

interface SelectorProps {
  languages: LocaleProperpties[]
}

const useStyles = makeStyles((theme) => ({
  formControl: {
    minWidth: 120,
  },
  item: {
    paddingLeft: '5px',
  },
}))

const Selector: React.SFC<SelectorProps> = ({ languages }) => {
  const classes = useStyles()
  const ctx = React.useContext(LocaleContext)
  const [language, setLanguage] = React.useState(ctx.locale)

  const handleChange = (event, setLocale) => {
    const path = event.target.value === 'en' ? '/' : `/${event.target.value}`
    setLanguage(event.target.value)
    setLocale(event.target.value)
    navigate(path)
  }

  const englishTranslation = (lang) => {
    if (lang.english_name) {
      return <span style={{ color: 'darkgrey' }}>({lang.english_name})</span>
    }
    return ''
  }

  return (
    <LocaleContext.Consumer>
      {({ locale, setLocale }) => (
        <FormControl className={classes.formControl}>
          <Select
            id="selector"
            value={language}
            onChange={(e) => handleChange(e, setLocale)}
            displayEmpty
            className={classes.selectEmpty}
            inputProps={{ 'aria-label': 'Without label' }}
          >
            {languages.map((lang) => (
              <MenuItem className={classes.item} key={lang.name} value={lang.path}>
                {lang.name}&nbsp; &nbsp;{englishTranslation(lang)}
              </MenuItem>
            ))}
          </Select>
          {/* <FormHelperText>Lanuage selector</FormHelperText> */}
        </FormControl>
      )}
    </LocaleContext.Consumer>
  )
}

export default Selector
