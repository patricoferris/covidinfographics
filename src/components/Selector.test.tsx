import React from 'react'
import renderer from 'react-test-renderer'
import Selector from './Selector'
import { shallow } from 'enzyme'

/* Fake data */
const locales = [
  {
    default: true,
    path: `en`,
    name: `English`,
    locale: `en-UK`,
    dateFormat: `DD/MM/YYYY`,
    siteLanguage: `en`,
    ogLanguage: `en_UK`,
    defaultTitle: `COVID19 Infographics`,
    defaultDescription: `The COVID19 website`,
  },
  {
    default: false,
    path: `fr`,
    name: `FranÃ§ais`,
    locale: `fr-FR`,
    dateFormat: `DD.MM.YYYY`,
    siteLanguage: `fr`,
    ogLanguage: `fr_FR`,
    defaultTitle: `Infographie de COVID19`,
    defaultDescription: `Le site de COVID19`,
  },
]

test('Clicking on selector reveals options for language', () => {
  const wrapper = shallow(<Selector languages={locales} />)
  expect(wrapper).toMatchSnapshot()
})
