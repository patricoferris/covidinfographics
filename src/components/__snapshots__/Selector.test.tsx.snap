// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clicking on selector reveals options for language 1`] = `
<WithStyles(ForwardRef(FormControl))
  className="makeStyles-formControl-1"
>
  <WithStyles(ForwardRef(Select))
    className="makeStyles-selectEmpty-2"
    displayEmpty={true}
    id="selector"
    inputProps={
      Object {
        "aria-label": "Without label",
      }
    }
    onChange={[Function]}
    value="en"
  >
    <WithStyles(ForwardRef(MenuItem))
      key="English"
      value="en"
    >
      English
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      key="Français"
      value="fr"
    >
      Français
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Select))>
  <WithStyles(ForwardRef(FormHelperText))>
    Lanuage selector
  </WithStyles(ForwardRef(FormHelperText))>
</WithStyles(ForwardRef(FormControl))>
`;
